File: src/Application/Client/Java/Quorum.java
package com.scalien.scaliendb;

import java.math.BigInteger;

/**
 * Quorum is a convenience class for working wit quorums.
 * <p>
 * Example:
 * <pre>
 * Quorum quorum = client.getQuorum("quorum 1");
 * // create table located in quorum
 * Database db = client.getDatabase("testDatabase");
 * db.createTable("testTable", quorum)
 * // list all quorums
 * for (Quorum q : client.getQuorums())
 *     System.out.println(q.Name);
 * </pre>
 * @see Client#getQuorum(String) 
 * @see Client#getQuorums() 
 * @see Database#createTable(String, Quorum) 
 */
public class Quorum
{
    private Client client;
    private long quorumID;
    private String name;
    
    Quorum(Client client, long quorumID, String name) throws SDBPException {
        this.client = client;
        this.quorumID = quorumID;
        this.name = name;
    }
    
    long getQuorumID() {
        return quorumID;
    }

    /**
     * Return the name of the quorum.
     */
    public String getName()
    {
        return name;
    }
}

File: src/Application/Client/CSharp/ScalienClient/Quorum.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace Scalien
{
    /// <summary>
    /// Quorum is a convenience class for working wit quorums.
    /// </summary>
    /// <example><code>
    /// Quorum quorum = client.GetQuorum("quorum 1");
    /// // create table located in quorum
    /// Database db = client.GetDatabase("testDatabase");
    /// db.CreateTable("testTable", quorum)
    /// // list all quorums
    /// foreach (Quorum q in client.GetQuorums())
    ///     System.Console.WriteLine(q.Name);
    /// </code></example>
    /// <seealso cref="Client.GetQuorum(string)"/>
    /// <seealso cref="Client.GetQuorums()"/>
    /// <seealso cref="Database.CreateTable(string, Quorum)"/>
    public class Quorum
    {
        private Client client;
        private ulong quorumID;
        private string name;

        public ulong QuorumID
        {
            get
            {
                return quorumID;
            }
        }

        /// <summary>
        /// The name of the quorum.
        /// </summary>
        public virtual string Name
        {
            get
            {
                return name;
            }
        }

        internal Quorum(Client client, ulong quorumID, string name)
        {
            this.client = client;
            this.quorumID = quorumID;
            this.name = name;
        }
    }
}
